src/Illuminate/Http/Concerns/InteractsWithInput.php

```diff
@@ -307,6 +308,33 @@ public function collect($key = null)
         return collect(is_array($key) ? $this->only($key) : $this->input($key));
     }
 
+    /**
+     * Retrieve input from the request as a Carbon instance.
+     *
+     * @param  string  $key
+     * @param  string|null  $format
+     * @param  string|null  $tz
+     * @return \Illuminate\Support\Carbon|bool|null
+     */
+    public function datetime($key, $format = null, $tz = null)
+    {
+        $date = $this->input($key);
+
+        if (is_null($date)) {
+            return null;
+        }
+
+        if ($format) {
+            try {
+                return Date::createFromFormat($format, $date, $tz);
+            } catch (InvalidArgumentException $e) {
+                return false;
```

Comment:
This should be `null`. This convenience method will only work if it returns either `\Illuminate\Support\Carbon` or `null`. Anything else will require additional type checks on the consumer side to ensure types are compatible (eg. when setting a model attribute to the return value of `$request->datetime()`).